src/worldfactory.h

```diff
@@ -2,11 +2,12 @@
 #ifndef WORLDFACTORY_H
 #define WORLDFACTORY_H
 
-#include "cursesdef.h"
 #include "enums.h"
 #include "json.h"
 #include "options.h"
 
+#include "UIHandler.h"
```

Comment:
A simple forward declaration is enough here, you don't actually need the whole header.